release failed `shred: /tmp/tmp.d4HYpg64Y8: failed to open for writing: No such file or directory`

run/19786001844/job/56695635169:
```
##[debug]Evaluating condition for step: 'Wait for NSG rule propagation and test SSH connectivity'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Wait for NSG rule propagation and test SSH connectivity
##[debug]Loading inputs
##[debug]Evaluating: format('# Create temporary SSH key file
##[debug]SSH_KEY_FILE=$(mktemp)
##[debug]trap ''shred -u "$SSH_KEY_FILE"'' EXIT
##[debug]echo "{0}" > "$SSH_KEY_FILE"
##[debug]chmod 600 "$SSH_KEY_FILE"
##[debug]
##[debug]echo "Testing SSH connectivity with up to 3 attempts..."
##[debug]for i in {{1..3}}; do
##[debug]  echo "Attempt $i/3: Testing SSH connection to {1}..."
##[debug]  if ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 -i "$SSH_KEY_FILE" deployuser@{2} "whoami"; then
##[debug]    echo "✓ SSH connection established"
##[debug]    shred -u "$SSH_KEY_FILE"
##[debug]    exit 0
##[debug]  fi
##[debug]  if [ "$i" -lt 3 ]; then
##[debug]    echo "SSH connection failed, waiting 30 seconds for NSG rule propagation..."
##[debug]    sleep 30
##[debug]  fi
##[debug]done
##[debug]echo "✗ SSH connection failed after 3 attempts (90s total)"
##[debug]shred -u "$SSH_KEY_FILE"
##[debug]exit 1
##[debug]', secrets.DEPLOY_SSH_KEY, steps.vm-ip.outputs.ip, steps.vm-ip.outputs.ip)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '# Create temporary SSH key file
##[debug]SSH_KEY_FILE=$(mktemp)
##[debug]trap ''shred -u "$SSH_KEY_FILE"'' EXIT
##[debug]echo "{0}" > "$SSH_KEY_FILE"
##[debug]chmod 600 "$SSH_KEY_FILE"
##[debug]
##[debug]echo "Testing SSH connectivity with up to 3 attempts..."
##[debug]for i in {{1..3}}; do
##[debug]  echo "Attempt $i/3: Testing SSH connection to {1}..."
##[debug]  if ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 -i "$SSH_KEY_FILE" deployuser@{2} "whoami"; then
##[debug]    echo "✓ SSH connection established"
##[debug]    shred -u "$SSH_KEY_FILE"
##[debug]    exit 0
##[debug]  fi
##[debug]  if [ "$i" -lt 3 ]; then
##[debug]    echo "SSH connection failed, waiting 30 seconds for NSG rule propagation..."
##[debug]    sleep 30
##[debug]  fi
##[debug]shred -u "$SSH_KEY_FILE"
##[debug]exit 1
##[debug]'
##[debug]Loading env
Run # Create temporary SSH key file
##[debug]/usr/bin/bash -e /home/runner/work/_temp/080060e6-02c0-4bf6-a842-dfd439f09c3b.sh
Testing SSH connectivity with up to 3 attempts...
Attempt 1/3: Testing SSH connection to 4.250.195.144...
Warning: Permanently added '4.250.195.144' (ED25519) to the list of known hosts.
deployuser
✓ SSH connection established
shred: /tmp/tmp.d4HYpg64Y8: failed to open for writing: No such file or directory
Error: Process completed with exit code 1.
##[debug]Finishing: Wait for NSG rule propagation and test SSH connectivity
```

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -193,7 +193,6 @@ jobs:
             echo "Attempt $i/3: Testing SSH connection to ${{ steps.vm-ip.outputs.ip }}..."
             if ssh -o StrictHostKeyChecking=no -o ConnectTimeout=10 -i "$SSH_KEY_FILE" deployuser@${{ steps.vm-ip.outputs.ip }} "whoami"; then
               echo "✓ SSH connection established"
-              shred -u "$SSH_KEY_FILE"
               exit 0
             fi
             if [ "$i" -lt 3 ]; then
@@ -202,7 +201,6 @@ jobs:
             fi
           done
           echo "✗ SSH connection failed after 3 attempts (90s total)"
-          shred -u "$SSH_KEY_FILE"
           exit 1
 
       - name: Deploy via SSH

